Felix5572 enable fep

felix5572 opened the pull request:
Unable LAMPPS  FEP package。 free energy calculation may involves this kind of pair. Especially for DeepMD force field.
@conda-forge-admin , please rerender

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [Y] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [Y] Bumped the build number (if the version is unchanged)
* [Y] Reset the build number to `0` (if the version changed)
* [Y] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [Y] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Unable LAMPPS  FEP package when compiling.
free energy calculation may involves this kind of pair. Especially for DeepMD force field.
@conda-forge-admin , please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12964793602](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12964793602). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12964825750](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12964825750). Examine the logs at this URL for more detail.</sub>

felix5572 commented:
@jan-janssen @njzjz
Hi, could you please review my PR? When using deepmd, especially when using the dpti (https://github.com/deepmodeling/dpti.git) software with deepmd-kit to calculate free energy, we need to use the lj/cut/soft pair style, which requires enabling the FEP package during LAMMPS compilation. I have now enabled this option during LAMMPS compilation.

njzjz commented:
```
2025-01-25T15:36:04.5948155Z OSError: libcuda.so.1: cannot open shared object file: No such file or directory
```

njzjz commented:
I think the tests for CUDA could be skipped, though I don't understand why there is no error before.

felix5572 commented:
@jan-janssen 


This PR resolves build failures introduced during the CUDA 12 migration by addressing Cython symbol conflicts in the ML-IAP package.
Changes Made
1. Build System Fixes (recipe/build.sh)

Linux: Added --allow-multiple-definition linker flag to suppress duplicate symbol errors
macOS: Disabled MLIAP_ENABLE_PYTHON to avoid Cython symbol conflicts (maybe no equivalent..)
Root cause: Duplicate __pyx_CommonTypesMetaclass_get_module symbols in:

mliap_unified_couple.cpp
mliap_model_python_couple.cpp


2. Runtime Testing Improvements (recipe/run_test.py)

Added GPU error handling for CI environments without GPU hardware
Graceful fallback when libcuda.so.1 is unavailable (expected in Azure CI)
Maintains test coverage while avoiding false failures
